server/app/rest_api/rest.py

```diff
@@ -150,10 +195,11 @@ def get(self):
         fields = request.args.getlist("annotation-name", None)
         try:
             annotation_response = current_app.data.annotation({}, "obs", fields)
+            return make_response(
+                annotation_response, HTTPStatus.OK, {"Content-Type": JSON_MIMETYPE}
+            )
         except KeyError:
             return make_response(f"Error bad key in {fields}", HTTPStatus.BAD_REQUEST)
-        try:
-            return make_response(jsonify(annotation_response), HTTPStatus.OK)
         except ValueError as e:
```

Comment:
One concern I have with handling ValueError at this level: it is a widely used Python error, and in the future could potentially be generated by many different code paths, thereby confusing JSON encoding errors with other conditions.

I think it would be better if this specific exception handling was wrapped directly around the call to JSON.dumps(), and then rethrown as an app-specific error (eg, JSONEncodingError)